src/java/org/apache/cassandra/index/sai/disk/format/IndexDescriptor.java

```diff
@@ -409,9 +409,9 @@ public void checksumPerIndexComponents(IndexTermType indexTermType, IndexIdentif
     }
 
     @SuppressWarnings("BooleanMethodIsAlwaysInverted")
-    public void checksumPerSSTableComponents()
+    public void checksumPerSSTableComponents(boolean validateChecksum)
```

Comment:
Looking at this, we could probably merge w/ `validatePerSSTableComponents()` above, and we'd get some extra logging for free. We'd have to pass an extra boolean directing the method to throw or not on `UncheckedIOException` though.

Also, looking at `validatePerSSTableComponents()` itself, the log message isn't specific enough. It should probably be:

```
logger.info(logMessage("Validating per-sstable index components for SSTable {} using mode {}..."), sstableDescriptor, validation);
```

Otherwise, the log message will be the same for every SSTable we validate :(

Reply:
CC @pkolaczk 

(Making sure I'm not missing something.)

Reply:
Either way, probably can't continue to start the method name w/ `checksum*`...

Reply:
@maedhroz `logMessage` in `validatePerSSTableComponents()` does print keyspace and table at least if you look into its implementation: 

    public String logMessage(String message)
    {
        // Index names are unique only within a keyspace.
        return String.format("[%s.%s.*] %s",
                             sstableDescriptor.ksname,
                             sstableDescriptor.cfname,
                             message);
    }

This however does not print Descrpitor's

    @Override
    public String toString()
    {
        return baseFile().absolutePath();
    }

We could include it into `logMessage` argument here.

Reply:
I have a hard time to understand what should be done here ...

Reply:
I simplified that.